src/test/java/org/opensearch/flowframework/FlowFrameworkRestTestCase.java

```diff
@@ -187,7 +203,7 @@ protected static void deleteIndexWithAdminClient(String name) throws IOException
 
     // Utility fn for checking if an index exists. Should only be used when not allowed in a regular context
     // (e.g., checking existence of system indices)
-    protected static boolean indexExistsWithAdminClient(String indexName) throws IOException {
+    public static boolean indexExistsWithAdminClient(String indexName) throws IOException {
```

Comment:
For my own education why do we need to use `adminClient()` here (and elsewhere) instead of `client()`?  

This method looks exactly like the superclass `indexExists()` method except using the `adminClient`.   The javadoc for the inherited `adminClient()` says it should only be used for cleanup (in the superclass it's only used with DELETE calls).   Is it because we're manipulating system indices?

Reply:
Yes, since this class supports both security enabled and disabled integration tests, in both cases, the admin client will always have permissions to search/delete system indices. As for why we're manipulating system indices, due to the node connectivity/drop issues we were seeing in the failed CI runs, I suspected the issue was lack of storage, so I thought deleting all created/populated indices after each test would mitigate this issue. 

On the same note, the reason why we exclude the ML config index from this clean up is because of the cron job that initializes the index and encryption key. If we delete it every single time, then each test in which we create a connector would need to wait for the next interval of the cron job to run in order to pass.